component/core/src/test/java/org/exoplatform/social/core/listeners/FileAttachmentListenerTest.java

```diff
@@ -0,0 +1,163 @@
+    fileAttachmentListener = new FileAttachmentListener(fileService);
+    if (metadataService.getMetadataTypeByName(METADATA_TYPE_NAME) == null) {
+      MetadataTypePlugin userMetadataTypePlugin = new MetadataTypePlugin(newParam(2659, METADATA_TYPE_NAME)) {
+        @Override
+        public boolean isAllowMultipleItemsPerObject() {
+          return true;
+        }
+
+        @Override
+        public boolean isShareable() {
+          return true;
+        }
+      };
+      metadataService.addMetadataTypePlugin(userMetadataTypePlugin);
+    }
+
+    johnIdentity = identityManager.getOrCreateUserIdentity("john");
+    tearDownActivityList = new ArrayList<>();
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    end();
+    begin();
+    identityManager.deleteIdentity(johnIdentity);
+    metadataDAO.deleteAll();
+
+    for (ExoSocialActivity activity : tearDownActivityList) {
+      activityManager.deleteActivity(activity.getId());
+    }
+    super.tearDown();
+  }
+
+  public void testDeleteActivity() throws Exception {
+    String activityTitle = "activity title";
+    String userId = johnIdentity.getId();
+    ExoSocialActivity activity = new ExoSocialActivityImpl();
+    activity.setTitle(activityTitle);
+    activity.setUserId(userId);
+    activityManager.saveActivityNoReturn(johnIdentity, activity);
+
+    long creatorId = Long.parseLong(johnIdentity.getId());
+    MetadataKey metadataKey = new MetadataKey(METADATA_TYPE_NAME, String.valueOf(FILE_ID), AUDIENCE_ID);
+    MetadataObject metadataObject = new MetadataObject(ExoSocialActivityImpl.DEFAULT_ACTIVITY_METADATA_OBJECT_TYPE,
+                                                       activity.getId(),
+                                                       PARENT_OBJECT_ID);
+    MetadataItem metadataItem = metadataService.createMetadataItem(metadataObject, metadataKey, creatorId);
+    assertNotNull(metadataItem);
+    assertTrue(metadataItem.getId() > 0);
+
+    List<MetadataItem> metadataItems = metadataService.getMetadataItemsByObject(metadataObject);
+    assertNotNull(metadataItem);
+    assertEquals(1, metadataItems.size());
+
+    FileItem fileItem = fileService.writeFile(new FileItem(FILE_ID,
+                                                           "test",
+                                                           "image/png",
+                                                           "attachment",
+                                                           "test".getBytes().length,
+                                                           new Date(),
+                                                           "user",
+                                                           false,
+                                                           new ByteArrayInputStream("test".getBytes())));
+
+    assertEquals(metadataItems.get(0).getMetadata().getType().getName(), METADATA_TYPE_NAME);
+
+    activityManager.deleteActivity(activity);
```

Comment:
You didn't tested what happens to attached files (it must be deleted automatically)
+
Avoid using `MetadataService` (low level Service + not concerned by the test) and use `AttachmentService` instead.

The Test should do This:
**When**:
1/ Create an activity
2/ Attach files into it (**Verify + add assertions that the files are really attached to the activity**)
(  + retrieve attached **file Ids** Using `AttachmentService` )
3/ Delete the activity

**Then**:
=> The previously attached files doesn't exist anymore